### PR TITLE
TfLite Atan2 missing datatype support

### PR DESCRIPTION
**Pull request by pratham-mcw**
-Adds bf16, f16 support for atan2
 -Adds bf16, f16 atan2 unit tests

### CODE REVIEW

File: tensorflow/lite/kernels/BUILD [hunk 1]
```diff
@@ -1944,7 +1944,7 @@ cc_test(
         "//tensorflow/lite/schema:schema_fbs",
         "@com_google_googletest//:gtest",
         "@flatbuffers",
-    ],
+    ]
```
Comment (rascani) [inline tensorflow/lite/kernels/BUILD:1947]: I think this change was unintentional, please remove.

File: tensorflow/lite/kernels/atan2.cc [hunk 2]
```diff
@@ -40,33 +38,31 @@ TfLiteStatus Atan2Prepare(TfLiteContext* context, TfLiteNode* node) {
...
   TfLiteTensor* output = tflite::GetOutput(context, node, 0);
-
   // Validate size and type constraints
   TF_LITE_ENSURE_OK(context, EnsureSameShape(context, input_y, input_x));
   TF_LITE_ENSURE_TYPES_EQ(context, input_y->type, input_x->type);
   TF_LITE_ENSURE_TYPES_EQ(context, input_y->type, output->type);
-  TF_LITE_ENSURE(context,
-                 input_y->type == kTfLiteFloat32 ||
-                 input_y->type == kTfLiteFloat64);
+  TF_LITE_ENSURE(context, input_y->type == kTfLiteFloat32 ||
```
Comment (rascani) [inline tensorflow/lite/kernels/atan2.cc:45]: Please make sure to run clang-format on this file, these seem misaligned.

File: tensorflow/lite/kernels/atan2_test.cc [hunk 3]
```diff
@@ -62,13 +69,77 @@ class Atan2Model : public tflite::SingleOpModel {
...
+
+template <>
+std::vector<Eigen::bfloat16> Atan2Model::GetOutput(
+    const std::vector<Eigen::bfloat16>& y,
+    const std::vector<Eigen::bfloat16>& x) {
+  PopulateTensor<Eigen::bfloat16>(y_, y);
+  PopulateTensor<Eigen::bfloat16>(x_, x);
+  Invoke();
+  return ExtractVector<Eigen::bfloat16>(output_);
+}
```
Comment (rascani) [inline tensorflow/lite/kernels/atan2_test.cc:89]: Why are these specializations necessary? They seem to implement the template exactly as default.

File: tensorflow/lite/kernels/atan2_test.cc [hunk 4]
```diff
@@ -94,16 +169,19 @@ TYPED_TEST(Atan2Test, TestBatch) {
...
   ASSERT_EQ(got.size(), 8);
   for (int i = 0; i < 8; ++i) {
-    EXPECT_FLOAT_EQ(got[i], std::atan2(y_data[i], x_data[i]));
+    EXPECT_FLOAT_EQ(got[i], Float(std::atan2(y_data[i], x_data[i])));
   }
 }
 
 }  // namespace
-}  // namespace tflite
+}  // namespace tflite
```
Comment (rascani) [inline tensorflow/lite/kernels/atan2_test.cc:187]: Please remove extra space at the end.

File: tensorflow/lite/kernels/atan2_test.cc [hunk 5]
```diff
@@ -62,13 +69,77 @@ class Atan2Model : public tflite::SingleOpModel {
...
 template <typename Float>
 class Atan2Test : public ::testing::Test {
  public:
   using FloatType = Float;
 };
 
-using TestTypes = ::testing::Types<float, double>;
+using TestTypes = ::testing::Types<float, double, Eigen::half, Eigen::bfloat16>;
+
+TEST(Atan2_test, Atan2_testWorks) {
```
Comment (rascani) [inline tensorflow/lite/kernels/atan2_test.cc:99]: What are these tests testing that is not covered under the typed tests? If just ensuring the type coverage works, lets rely on the typed tests for that. If something else, lets use better names for these test cases so they convey the behavior.

**Review by rascani — CHANGES_REQUESTED** [review]

File: tensorflow/lite/kernels/BUILD (hunk 1 above)
Comment (pratham-mcw) [reply tensorflow/lite/kernels/BUILD]: ok

File: tensorflow/lite/kernels/atan2.cc (hunk 2 above)
Comment (pratham-mcw) [reply tensorflow/lite/kernels/atan2.cc:45]: Okay

File: tensorflow/lite/kernels/atan2_test.cc (hunk 3 above)
Comment (pratham-mcw) [reply tensorflow/lite/kernels/atan2_test.cc]: Yes those weren't needed and I have made the changes.

File: tensorflow/lite/kernels/atan2_test.cc (hunk 4 above)
Comment (pratham-mcw) [reply tensorflow/lite/kernels/atan2_test.cc]: Okay

File: tensorflow/lite/kernels/atan2_test.cc (hunk 5 above)
Comment (pratham-mcw) [reply tensorflow/lite/kernels/atan2_test.cc]: I've changed it to typed test

**Review by rascani — APPROVED** [review]

**Pull request merged**